sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -535,6 +535,14 @@ class HiveContext private[hive](
     ).map { case (confVar, unit) =>
       confVar.varname -> hiveconf.getTimeVar(confVar, unit).toString
     }.toMap
+
+    // SPARK-11783 When "hive.metastore.uris" is set, it overrides "javax.jdo.option.ConnectionURL",
+    // thus the execution Hive client connects to the actual remote Hive metastore instead of the
+    // Derby metastore created in the temporary directory.  Cleaning this configuration for
+    // the execution Hive client fixes this issue.
+    timeConfVars ++ Seq(
+      ConfVars.METASTOREURIS.varname -> ""
```

Comment:
This `configure()` is only used by `metadataHive`, I'm wondering how this patch could fix the problem?